cypress/integration/join.spec.js

```diff
@@ -110,134 +165,403 @@ describe('join', () => {
+    cy.get('button[type="submit"]').click();
+
+    cy.url().should('contain', '/join');
+
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.getCookies().should('have.length', 0);
+  });
+
+  it('should NOT be able to register when firstName contains only spaces', () => {
+    cy.get(inputFields.email).type(validUser.email);
+    cy.get(inputFields.confirmEmail).type(validUser.email);
+    cy.get(inputFields.password).type(validUser.password);
+    cy.get(inputFields.confirmPassword).type(validUser.password);
+
+    cy.get(inputFields.firstName).type('     ').blur();
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.get(inputFields.lastName).type(validUser.lastName);
+    cy.get(inputFields.zipcode).type(validUser.zipcode);
+    cy.get('button[type="submit"]').click();
+
+    cy.url().should('contain', '/join');
+
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.getCookies().should('have.length', 0);
+  });
+
+  it('should NOT be able to register when blurring past lastName', () => {
+    cy.get(inputFields.email).type(validUser.email);
+    cy.get(inputFields.confirmEmail).type(validUser.email);
+    cy.get(inputFields.password).type(validUser.password);
+    cy.get(inputFields.confirmPassword).type(validUser.password);
+    cy.get(inputFields.firstName).type(validUser.firstName);
+
+    cy.get(inputFields.lastName).focus().blur();
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.get(inputFields.zipcode).type(validUser.zipcode);
+    cy.get('button[type="submit"]').click();
+
+    cy.url().should('contain', '/join');
+
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.getCookies().should('have.length', 0);
+  });
+
+  it('should NOT be able to register when lastName contains only spaces', () => {
+    cy.get(inputFields.email).type(validUser.email);
+    cy.get(inputFields.confirmEmail).type(validUser.email);
+    cy.get(inputFields.password).type(validUser.password);
+    cy.get(inputFields.confirmPassword).type(validUser.password);
+    cy.get(inputFields.firstName).type(validUser.firstName);
+
+    cy.get(inputFields.lastName).type('     ').blur();
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.get(inputFields.zipcode).type(validUser.zipcode);
+    cy.get('button[type="submit"]').click();
+
+    cy.url().should('contain', '/join');
+
+    cy.get('div[role="alert"]')
+      .should('have.length', 1)
+      .should('contain', validationErrorMessages.required);
+
+    cy.getCookies().should('have.length', 0);
```

Comment:
It looks like you re-use this block of logic many times.

Maybe you could make a function like:

```js
const assertFailedLogin = ({ numberOfErrors }) => {
  cy.url().should('contain', '/join');
  cy.getCookies().should('have.length', 0);

  cy.get('div[role="alert"]')
    .should('have.length', numberOfValidationErrors)
    .should('contain', validationErrorMessages.required);
}

// ...

assertFailedLogin({ numberOfErrors: 1 });
```